tests/core/filtering/test_filters_against_many_blocks.py

```diff
@@ -32,20 +32,28 @@ def single_transaction(w3):
     return tx_hash
 
 
+@pytest.mark.parametrize('api_style', ('v3', 'v4', 'build_filter'))
 def test_event_filter_new_events(
         web3,
         emitter,
         Emitter,
         wait_for_transaction,
         emitter_event_ids,
+        api_style,
         create_filter):
 
     matching_transact = emitter.functions.logNoArgs(
         which=1).transact
     non_matching_transact = emitter.functions.logNoArgs(
         which=0).transact
 
-    event_filter = emitter.events.LogNoArguments().createFilter(fromBlock='latest')
+    if api_style == 'build_filter':
+        builder = emitter.events.LogNoArguments.build_filter()
+        builder.fromBlock = 'latest'
+        builder.address = emitter.address
```

Comment:
address should be filled in when creating the builder object from a deployed contract.

Reply:
I'm not sure I understand?  As a user, I would be surprised if the address was not automatically populated when I created a filter from a contract instance.

Reply:
Just making a note to myself to fix this.